cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -339,16 +340,14 @@ class SnappyUnifiedMemoryManager private[memory](
   }
 
   private def getMinHeapEviction(required: Long): Long = {
-    // evict at least 100 entries to reduce GC cycles
     val waitingThreads = threadsWaitingForStorage.get()
-    math.max(required * waitingThreads, math.min(minHeapEviction,
-      required * math.max(100L, waitingThreads + 1)))
+    math.max(required, math.min(math.max(minHeapEviction, required * waitingThreads),
```

Comment:
Just to understand: I think with the previous code there was a possibility of this function returning more than maxHeapEviction i.e. 1 GB. Right?
Also, I see that @rishitesh idea was to keep 100 factor to simulate at least 100 entry eviction at once ( assuming number of waiting threads is much less and required * 100 is less than 10 MB) but that is gone now. I guess he did this so that smaller eviction does not happen too often. But here again that chance is created for the same.

Reply:
This change of 100 factor was by me. I am still thinking of all pros and cons here. Will revert the heap change for now and come back to it later and with more testing.